[Improvement] - Converted store to c++, refactor all code and fix some related bugs

dudantas opened the pull request:
Added store in c++ and removed store in modules.

Still needs testing, there are some minor tweaks that need to be done.

Known bugs: Uncategorized offers appearing last.

dudantas committed:
Gamestore in cpp

dudantas committed:
Gamestore functions and bytes

dudantas committed:
Delete client_assertions.txt

dudantas committed:
Fix indent

dudantas committed:
Store loading informations done

dudantas committed:
Store working 100%

dudantas committed:
Fix bug in the "Account::AddCoins" and "Account::RemoveCoins" functions and add new functions

dudantas committed:
Fix bug add/remove//get store coins and tournament store coins script

dudantas committed:
Fix bug in store coins transfer

dudantas committed:
Move store coins talkactions to store folder

dudantas committed:
Fix not receiving outfit bug

dudantas committed:
Comment to explain the function

Costallat commented on src/creatures/creatures_definitions.hpp:
```diff
@@ -21,6 +21,18 @@
+	int32_t cust;
+
+};
 
```
```suggestion
struct StoreHistory {
	StoreHistory(uint32_t time, uint8_t mode, uint32_t amount, uint8_t coinMode, std::string description, int32_t cust) : 
		time(time), mode(mode), amount(amount), coinMode(coinMode), description(std::move(description)), cust(cust) {}

	uint32_t time;
	uint8_t mode;
	uint32_t amount;
	uint8_t coinMode;
	std::string description;
	int32_t cust;

};

// Enum
```

dudantas committed:
Somes small fixes and adjustments

dudantas committed:
More fixes

dudantas committed:
Fix compilation error

Costallat commented on src/creatures/players/account/account.cpp:
```diff
@@ -65,161 +72,162 @@ Account::Account(const std::string &email) : email_(email) {
  ******************************************************************************/
 
-error_t Account::GetCoins(uint32_t *coins) {
+error_t Account::GetCoins(CoinType_t coinType) {
```
keep old function and create a new GetTournamentCoins

Costallat commented on src/creatures/players/account/account.cpp:
```diff
@@ -65,161 +72,162 @@ Account::Account(const std::string &email) : email_(email) {
-
-  uint32_t current_coins = 0;
-  this->GetCoins(&current_coins);
+error_t Account::RemoveCoins(int32_t amount, CoinType_t coinType) {
```
Same here, keep old function and create RemoveTournamentCoins

Costallat commented on src/creatures/players/account/account.cpp:
```diff
@@ -65,161 +72,162 @@ Account::Account(const std::string &email) : email_(email) {
-          " `description`) VALUES (" << id_ << ", " << static_cast<uint16_t>(type) << ", "<< coins
-          << ", " << db_->escapeString(description) << ")";
+	std::ostringstream query;
+	query << "INSERT INTO `store_history` (`account_id`, `time`, `mode`, `amount`, `coinMode`, `description`, `cust`) VALUES ("
```
Here the idea is to be generic to store all coins transactions, not only for the store, so if there are something new that use the coins, we don't need to create a new table

Costallat commented on src/creatures/players/player.cpp:
```diff
@@ -4612,9 +4612,62 @@ void Player::setPremiumDays(int32_t v)
 }
 
-void Player::setTibiaCoins(int32_t v)
+void Player::setStoreCoins(int32_t coins, CoinType_t coinType)
```
Try to break in setCoins / setTournamentCoin

dudantas replied:
Great suggestion, I'll do it.

Costallat commented on src/creatures/players/store/store.hpp:
```diff
@@ -0,0 +1,354 @@
+
+class StoreOffers {
+	public:
+		StoreOffers(std::string name) : 
```
```suggestion
		explicit StoreOffers(std::string name) : 
```

dudantas replied:
Thanks

dudantas committed:
Update src/creatures/creatures_definitions.hpp

Co-authored-by: Renato Foot Guimarães Costallat <costallat@hotmail.com>

dudantas committed:
Update src/creatures/players/store/store.hpp

Co-authored-by: Renato Foot Guimarães Costallat <costallat@hotmail.com>

dudantas committed:
Moved the premium discount rate in the store to config.lua, and other fixes

dudantas committed:
Set discount store to false

dudantas committed:
Add vcpkg manifest file

dudantas committed:
Fix: issue not displaying description in store offers

dudantas committed:
Separation of coins and tournament coins

The coin and the tournament coin were uncoupled, just as @costallat asked in the review. Thanks!
Fixed wrong ids from store.xml

Costallat commented on data/scripts/talkactions/god/store/coins.lua:
```diff
@@ -0,0 +1,169 @@
+		return false
+	end
+
+	player:sendTextMessage(MESSAGE_EVENT_ADVANCE, "Player ".. targetPlayer:getName() .." have \z
```
```suggestion
	player:sendTextMessage(MESSAGE_EVENT_ADVANCE, "Player ".. targetPlayer:getName() .." have " \z
```

Costallat commented on data/scripts/talkactions/god/store/tournament_coins.lua:
```diff
@@ -0,0 +1,171 @@
+		return false
+	end
+
+	player:sendTextMessage(MESSAGE_EVENT_ADVANCE, "Player ".. targetPlayer:getName() .." \z
+                           have ".. targetPlayer:getTournamentCoins() .." tournament coins.")
```
```suggestion
	player:sendTextMessage(MESSAGE_EVENT_ADVANCE, "Player ".. targetPlayer:getName() .. \z
                           " have ".. targetPlayer:getTournamentCoins() .." tournament coins.")
```

dudantas committed:
Fix identations

Costallat committed:
Refactor account class

Signed-off-by: Renato Foot <costallat@hotmail.com>

Costallat committed:
Update account documentation

Signed-off-by: Renato Foot <costallat@hotmail.com>

Costallat committed:
Fix windows build

Signed-off-by: Renato Foot <costallat@hotmail.com>

Costallat committed:
Adjust code style

Signed-off-by: Renato Foot <costallat@hotmail.com>

beats-dh committed:
Merge branch 'master' into gamestore-cpp

beats-dh committed:
init storegame.lua > store.xml

Co-Authored-By: guispiller <11639062+guispiller@users.noreply.github.com>

beats-dh committed:
fixs store.xml

dudantas committed:
Merge branch 'master' into gamestore-cpp

dudantas committed:
Remove whitespace
